packages/svelte/preprocess.js

```diff
@@ -27,15 +27,21 @@ const preprocess = options => ({
          */
         let code = content
             .replace(
-                /<script(.*?)>(.*)<\/script>/ms,
-                (match, attributes, content) => {
-                    script = {attributes, content};
+                /<script(.*?)>(.*?)<\/script>/gms,
+                (_, attributes, content) => {
+                    if (attributes.includes(`context="module"`)) {
+                        script.content += content;
```

Comment:
I'm not sure, that we should put the content of the `context="module"` script to the common script.
it looks like after preprocess this code:
```html
<script>...</script>
<script context="module">...</script>
```
will be transformed to
```html
<script>...</script>
```

which is not correct, because the code on `context="module"` has a different meaning, as I understood the docs.

so maybe to do the thing like this?

```js
const moduleScript = ''

... (match, attributes, content) => {
    if (attributes.includes(`context="module"`)) {
        moduleScript = match
    } else {
        ...
    }
}

let result = `${moduleScript}<script${script.attributes}>${code}</script>`;
...

```

Reply:
It is also possible to define variables in a `module script` so it should also be parsed.
Ex:
```html
<script context="module">
    let newColor = 'blue';
</script>

<script>
    let color = 'tomato';
</script>

<style reshadow>
    button {
        padding: 5px 10px;
        border-radius: 5px;
        font-size: 16px;
    }

    button[|type='primary'] {
        color: green;
    }

    button[|size='s'] {
        font-size: 14px;
        color: val(color);
    }

    button[|size='m'] {
        font-size: 16px;
        color: val(newColor);
    }
</style>

<button :size={'s'}>Small</button>
<button :size={'m'}>Medium</button>
<button :type={'primary'}>Primary</button>
```

Reply:
`module script` runs at the first component instance, `script` tags are evaluated at every component instance.

Reply:
Usually `module script` is used for fetching data and animations.

Reply:
thank you very much for the clarification!

anyway it looks like that we should keep `module script` in the separate `script` and don't merge it with the main `script`?

and if we will restrict the usage of `css-in-js` in the `module script` (I mean, no `styled` function usage there), thus we don't need to parse its code, because links to the variables will be anyway proceed, for example:

```js
<script context="module">
    let newColor = 'blue';
</script>

<script>
    let color = 'tomato';

    // ...
     $: __styles__ = __styled__(
        (set([__styled___c8], `--c8_0:${color};--c8_1:${newColor};`),
        __extract__()),
    );
</script>

```

Reply:
That makes sense, so Reshadow won't be used inside `script module` but it will still be able to access the variables through Svelte runtime.

Reply:
yep, exactly